pkg/network/telemetry/utils.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/common"
+)
+
+func contains(want string, strings []string) bool {
+	for _, candidate := range strings {
+		if want == candidate {
+			return true
+		}
+	}
+	return false
+}
+
+// Example: given {"_opt3", "tag:a", "_opt2", "tag:b", "_opt1", "tag:c"}
+// The return value will be:
+// {"tag:a", "tag:b", "tag:c"}
+// {"_opt1", "_opt2", "_opt3"}
+func splitTagsAndOptions(all []string) (tags, opts []string) {
+	if len(all) == 0 {
+		return
+	}
+	tagSet := common.NewStringSet()
+	optSet := common.NewStringSet()
+
+	for _, s := range all {
+		if strings.HasPrefix(s, optPrefix) {
+			optSet.Add(s)
+		} else {
+			tagSet.Add(s)
+		}
+	}
+
+	tags = tagSet.GetAll()
+	opts = optSet.GetAll()
+	sort.Strings(tags)
+	sort.Strings(opts)
+
+	return
+}
+
+func insertNestedValueFor(name string, value int64, root map[string]interface{}) {
```

Comment:
seems like a very specific assumption here about `root` parameter
please document the function so it will be clear the usage and the assumptions

Reply:
Sounds good :+1: